include/dlaf/communication/init.h

```diff
@@ -26,6 +31,12 @@ namespace comm {
 struct mpi_init {
   /// Initialize MPI to MPI_THREAD_MULTIPLE
   mpi_init(int argc, char** argv) noexcept {
+    // On older Cray MPICH versions initializing HIP after MPI leads to HIP not seeing any devices. Hence
```

Comment:
Do you know which version is ok and which is problematic?
It would be nice to put it in the comment.